File: Backend/model/src/main/java/io/sparkled/model/animation/param/ParamName.java
package io.sparkled.model.animation.param;

public enum ParamName {
    COLOR("Color"),
    CYCLE_COUNT("Cycle Count"),
    CYCLES_PER_SECOND("Cycles Per Second"),
    LENGTH("Length"),
    PERCENT("Percent");

    private String name;

    ParamName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }
}

File: model/src/main/kotlin/io/sparkled/model/animation/param/ParamName.kt
package io.sparkled.model.animation.param

enum class ParamName constructor(val displayName: String) {
    NONE("None"),
    BLEND_HARDNESS("Blend Hardness (%)"),
    BRIGHTNESS("Brightness (%)"),
    DENSITY("Density (%)"),
    COLOR("Color"),
    COLORS("Colors"),
    CYCLE_COUNT("Cycle Count"),
    CYCLES_PER_SECOND("Cycles Per Second"),
    LENGTH("Length"),
    LIFETIME("Lifetime (Seconds)"),
    PERCENT("Percent"),
    RANDOM_SEED("Random Seed"),
}
